spec/active_job/queue_adapters/sidekiq_publisher_adapter_spec.rb

```diff
@@ -14,38 +16,99 @@ def perform(*args); end
 
   before do
     stub_const("TestJob", job_class)
+    clear_redis
   end
 
   describe "#enqueue" do
-    it "creates a SidekiqPublisher job" do
-      active_job.enqueue
+    context "when in a transaction" do
+      it "creates a SidekiqPublisher job record" do
```

Comment:
❓ Think it's worth verifying here that we don't push to Redis and that we don't create a record when we're not in a transaction? Not sure if it's worth it in every test group or not but might be nice to have at least once.

Reply:
+1

Reply:
The `expect(job).to be_nil` expectations verify a record wasn't created. But yeah, when in a transaction, we can also verify it doesn't write to Redis too.

Reply:
Ah, I think I missed that because I was reading the test description and not the test itself.

Reply:
Yeah, and that line isn't very clear. I separated those two expectations into separate specs in [8d7c250](https://github.com/ezcater/sidekiq_publisher/pull/74/commits/8d7c250c1e04c10bdd6281666aea7b507f32cdd0).